romeo/src/actor.rs

```diff
@@ -0,0 +1,171 @@
+use std::time::Duration;
+
+use futures::Future;
+use serde::{de::DeserializeOwned, Serialize};
+use tokio::{
+    sync::broadcast::{self, error::RecvError, Sender},
+    task::JoinHandle,
+    time::sleep,
+};
+
+use crate::event::Event;
+use crate::store::Store;
+
+pub trait Actor: Serialize + DeserializeOwned + Send + Sync + 'static {
+    const NAME: &'static str;
+
+    fn handle(&mut self, event: Event) -> anyhow::Result<Vec<Event>>;
+}
+
+pub struct System<S> {
+    store: S,
+    sender: Sender<Event>,
+    handles: Vec<JoinHandle<()>>,
+}
+
+impl<S> System<S> {
+    pub fn new(store: S) -> Self {
+        let (sender, _) = broadcast::channel(128);
```

Comment:
Yeah, the actors are independent of `tokio`, whereas the outside system uses it. Since we'll be using a lot of I/O I think it makes sense to use `tokio`, but I want to keep it as isolated from the business logic as possible.